v2_task.go

```diff
@@ -22,6 +24,36 @@ type TaskDateTime struct {
 	time.Time
 }
 
+type AmDuration string
+
+type Task struct {
+	ID          string       `json:"uuid"`
+	ExitCode    uint8        `json:"exit_code"`
+	FileID      string       `json:"file_uuid"`
+	Filename    string       `json:"file_name"`
+	CreatedAt   TaskDateTime `json:"time_created"`
+	StartedAt   TaskDateTime `json:"time_started"`
+	CompletedAt TaskDateTime `json:"time_ended"`
+	Duration    AmDuration   `json:"duration"`
+}
+
+func (t *AmDuration) UnmarshalJSON(data []byte) error {
+	s := string(data)
+	s = s[1 : len(s)-1]
+	ok, err := regexp.Match("[<>] ", data)
+	if !ok {
+		_, err = strconv.ParseUint(s, 10, 32)
+	} else {
+		_, err = strconv.ParseUint(s[2:], 10, 32)
+	}
+	if err != nil {
+		return err
+	}
+	*t = AmDuration(s)
+
+	return err
+}
```

Comment:
You don't want to use regexp in this code path because of the performance implications. We can get away with some basic byte slice comparisons.

I've written an alternative here including tests: https://go.dev/play/p/Ej1aF6Zv-FC. Perhaps you can include my code in `duration.go` and `duration_test.go`.

Reply:
I made some slight changes. I really like and appreciate the work that went into these suggestions. I was considering benchmarking the two as well. Will do that later!